scaffold/cmd/source.go

```diff
@@ -57,44 +44,47 @@ type scaffoldData struct {
 }
 
 func runScaffoldSource(org string, name string, outputDir string) error {
-	tpl, err := template.New("source").ParseFS(sourceFS,
-		"templates/source/.github/workflows/*.tpl",
-		"templates/source/*.tpl",
-		"templates/source/plugin/*.tpl",
-		"templates/source/client/*.tpl",
-		"templates/source/resources/*.tpl",
-	)
-	if err != nil {
-		return fmt.Errorf("failed to parse templates: %w", err)
-	}
-
 	data := scaffoldData{
 		Org:  org,
 		Name: name,
 	}
-	for templatePath, filePath := range scaffoldTemplates {
-		var sb strings.Builder
-		if err := tpl.ExecuteTemplate(&sb, templatePath, data); err != nil {
-			return fmt.Errorf("failed to execute template: %w", err)
+
+	return fs.WalkDir(sourceFS, "templates/source", func(templatePath string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to walk directory: %w", err)
 		}
-		content := []byte(sb.String())
-		fullPath := outputDir + "/" + filePath
-		baseDir := path.Dir(fullPath)
-		if err := os.MkdirAll(baseDir, 0755); err != nil {
-			return fmt.Errorf("failed to create directory %s: %w", baseDir, err)
+		if d.IsDir() {
+			return nil
 		}
-		if strings.HasSuffix(filePath, ".go") {
-			formattedContent, err := format.Source(content)
+		if strings.HasSuffix(templatePath, ".tpl") {
+			tpl, err := template.New(filepath.Base(templatePath)).ParseFS(sourceFS, templatePath)
```

Comment:
Mostly the same code as before, and now getting the output path based on the template filename